src/main/java/io/harness/cf/client/api/StorageRepository.java

```diff
@@ -49,6 +56,37 @@ public Optional<FeatureConfig> getFlag(@NonNull String identifier) {
     return getFlag(identifier, true);
   }
 
+  public List<String> getAllFeatureIdentifiers(String prefix) {
+    List<String> identifiers = new LinkedList<>();
+    List<String> keys = cache.keys();
+    String flagPrefix = "flags/";
+    for (String key : keys) {
+      if (key.startsWith(flagPrefix)) {
+        // Strip the flag prefix
+        String strippedKey = key.substring(flagPrefix.length());
+        // If prefix is empty, add all stripped keys, otherwise check for prefix match
+        if (prefix.isEmpty() || strippedKey.startsWith(prefix)) {
+          identifiers.add(strippedKey);
+        }
+      }
+    }
+    return identifiers;
+  }
+
+  public Optional<FeatureConfig[]> getCurrentAndPreviousFeatureConfig(@NonNull String identifier) {
+    final String flagKey = formatFlagKey(identifier);
+    final String pFlagKey = formatPrevFlagKey(identifier);
+
+    FeatureConfig pFlag = (FeatureConfig) cache.get(pFlagKey);
+    FeatureConfig cFlag = (FeatureConfig) cache.get(flagKey);
+
+    // we should have at least current flag there to return.
+    if (cFlag != null) {
```

Comment:
Good check.  I am wondering what if the "previous" flag isn't there - do we need to also check it for null before adding it to the `FeatureConfig[]` array?